File: 卡码/代码随想录/JS/动态规划/02-背包问题/02-完全背包/LeetCode_0322.js
//322. 零钱兑换
const coinChange = function(coins, amount) {
	const dp=new Array(amount+1).fill(Number.MAX_VALUE-1);
	dp[0]=0;
	for(let coin of coins){
		for(let j=coin;j<=amount;j++){
			dp[j]=Math.min(dp[j],dp[j-coin]+1);
		}
	}
	return dp[amount]==Number.MAX_VALUE-1 ? -1 : dp[amount];
};

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/动态规划/_02_背包问题/_02_完全背包/LeetCode_0322.java
package com.kurumi.leetCode.codeThoughts.动态规划._02_背包问题._02_完全背包;

import java.util.Arrays;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 322. 零钱兑换
 */
public class LeetCode_0322 {
    static class Solution {
        public int coinChange(int[] coins, int amount) {
            int[] dp = new int[amount + 1];
            Arrays.fill(dp, Integer.MAX_VALUE - 1);
            dp[0] = 0;
            for (int coin : coins) {
                for (int j = coin; j <= amount; j++) {
                    dp[j] = Integer.min(dp[j], dp[j - coin] + 1);
                }
            }
            return dp[amount] == Integer.MAX_VALUE - 1 ? -1 : dp[amount];
        }
    }
}
